streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -237,10 +247,79 @@ public final TopologyBuilder addProcessor(String name, ProcessorSupplier supplie
 
         nodeNames.add(name);
         nodeFactories.add(new ProcessorNodeFactory(name, parentNames, supplier));
+        nodeGroups.add(name);
+        nodeGroups.unite(name, parentNames);
         return this;
     }
 
     /**
+     * Returns the topic groups.
+     * A topic group is a group of topics in the same task.
+     *
+     * @return groups of topic names
+     */
+    public Collection<Set<String>> topicGroups() {
```

Comment:
I feel we can consolidate ```topicGroups``` with ```copartitionGroups``` since two topics not within the same topology (i.e. the former group) will never be in the latter group. But we can do this in another PR moving forward.